src/Agent.Listener/Program.cs

```diff
@@ -19,6 +20,30 @@ public static int Main(string[] args)
 
         public async static Task<int> MainAsync(string[] args)
         {
+#if OS_LINUX
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                Console.WriteLine("This Agent version is built for Linux. Please download a corrent build for your OS.");
```

Comment:
I'll try to localize these strings. I'm not sure if we want to exit immediately, because localization may require to run some initialization code, which may assume we are on the correct OS (for example line endings are different).